tool/site/client/src/pages/claimPage.tsx

```diff
@@ -83,11 +93,96 @@ async function retrievePrescriptionDetails(baseUrl: string, prescriptionId: stri
     throw new Error("Dispense notification not found. Has this prescription been dispensed?")
   }
 
-  return {
+  const prescriptionDetails = {
     patient: getPatientResources(prescriptionOrder)[0],
     medicationRequests: getMedicationRequestResources(prescriptionOrder),
     medicationDispenses: dispenseNotifications.flatMap(getMedicationDispenseResources)
   }
+
+  if (amend) {
+    const claimResponse = await axiosInstance.get<fhir.Claim>(`${baseUrl}claim/${prescriptionId}`)
+    const claim = getResponseDataIfValid(claimResponse, isClaim)
+    return {
+      ...prescriptionDetails,
+      claim
+    }
+  }
+
+  return prescriptionDetails
+}
+
+export function getInitialValues(products: Array<StaticProductInfo>, previousClaim?: fhir.Claim): ClaimFormValues {
+  if (previousClaim) {
+    const productInfo = getProductInfo(products, previousClaim)
+    const exemptionInfo = getExemptionInfo(previousClaim)
+    return {
+      products: productInfo,
+      exemption: exemptionInfo
+    }
+  } else {
+    const defaultValues = {
+      products: products.map(product => ({
+        ...product,
+        patientPaid: false,
+        endorsements: []
+      })),
+      exemption: {
+        code: PRESCRIPTION_CHARGE_EXEMPTION_CODE_NONE,
+        evidenceSeen: false
+      }
+    }
+    return defaultValues
+  }
+}
+
+function getProductInfo(products: Array<StaticProductInfo>, previousClaim: fhir.Claim): Array<ProductFormValues> {
+  const claimDetails = previousClaim.item.flatMap(item => item.detail)
+  return claimDetails.map(detail => {
+    const claimDetailIdentifierExtension = getClaimMedicationRequestReferenceExtension(detail.extension)
+    const claimDetailIdentifier = claimDetailIdentifierExtension.valueReference.identifier.value
+
+    const associatedProduct = products.find(product => product.id === claimDetailIdentifier)
+
+    return {
+      ...associatedProduct,
+      patientPaid: getPatientPaid(detail),
+      endorsements: getEndorsementInfo(detail)
+    }
+  })
+}
+
+function getEndorsementInfo(detail: fhir.ClaimItemDetail): Array<EndorsementFormValues> {
+  const endorsementCodeableConcepts = detail.programCode
+    .filter(codeableConcept => codeableConcept.coding
+      .some(coding => coding.system === "https://fhir.nhs.uk/CodeSystem/medicationdispense-endorsement"))
+  return endorsementCodeableConcepts.map(codeableConcept => ({
+    code: codeableConcept.coding.find(coding => coding.system === "https://fhir.nhs.uk/CodeSystem/medicationdispense-endorsement").code,
+    supportingInfo: codeableConcept?.text
+  }))
+}
+
+function getPatientPaid(detail: fhir.ClaimItemDetail): boolean {
```

Comment:
Are there helper functions for this like getCodeableConceptFromUrl or something?